### PR TITLE
Move the checks for Arguments constructors to inline const

### PR DESCRIPTION
**Pull request by saethlin**
Thanks @Skgland for pointing out this opportunity: https://github.com/rust-lang/rust/pull/117804#discussion_r1612964362

### CODE REVIEW

File: library/core/src/fmt/mod.rs [hunk 1]
```diff
@@ -338,22 +338,22 @@ pub struct Arguments<'a> {
...
-            crate::panicking::panic_nounwind("invalid args");
+    pub fn new_v1<const P: usize, const A: usize>(
+        pieces: &'a [&'static str; P],
+        args: &'a [rt::Argument<'a>; A],
+    ) -> Arguments<'a> {
+        const {
+            if P < A || P > A + 1 {
+                panic!("invalid args");
+            }
         }
```
Comment (fmease) [inline library/core/src/fmt/mod.rs:357]:
```suggestion
        const { assert!(P >= A && P <= A + 1, "invalid args") }
```

**Comment by saethlin** [comment]
r? libs

**Comment by joboet** [comment]
Very cool!
@bors r+
r? joboet

**Pull request merged**